enhancements/network/admin-network-policy.md

```diff
@@ -0,0 +1,322 @@
+  - npinaeva, OpenShift Networking
+approvers:
+  - trozet, OpenShift Networking
+api-approvers:
+  - None
+creation-date: 2023-05-31
+last-updated: 2023-06-1
+tracking-link:
+  - https://issues.redhat.com/browse/SDN-2931
+  - https://issues.redhat.com/browse/SDN-2932
+see-also:
+  - "https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2091-admin-network-policy"
+replaces:
+  - None
+superseded-by:
+  - None
+---
+
+# Add Support for Admin Network Policy in OVN-Kubernetes
+
+## Summary
+
+This enhancement outlines the design details to implement `Admin Network Policy` and
+`Baseline Admin Network Policy` APIs in OVN-Kubernetes CNI plugin. These are
+[new APIs that are currently in v1alpha1 version](https://github.com/kubernetes-sigs/network-policy-api/tree/master/apis/v1alpha1)
+and maintained by upstream Kubernetes `sig-network-policy-api` working group.
+
+## Motivation
+
+Today, OVN-Kubernetes supports Kubernetes Network Policy API that is being used
+by all personas (administrators, operators, developers, application owners, tenant admins) for enforcing network traffic regulations on their cluster worloads.
+However Network Policies are namespace-scoped and were designed for developers
+or namespace owners. They don't satisfy the requirements or use cases for cluster administrators. Some of the main motivations for ANP and BANP are:
+
+1. Cluster administrators don't have a way to define network policies on a
+cluster-scoped level that is non-overridable by network policies.
+
+2. Cluster administrators want network policies to be in place before the
+namespaces are created.
+
+3. Network Policies use an "implicit isolation" model that creates a default
+deny ACL as soon as the policy is applied to the workload, thus automatically
+isolating the workload. Any traffic that needs to be allowed, has to be called
+out explicitly. For network administrators, they prefer an allowList and DenyList
+that is explicit and an API that is similar to their traditonal firewall rules
+that doesn't make any implicit assumptions.
+
+### User Stories
+
+* > "As an OpenShift cluster administrator I want to apply cluster-scoped policies
+and guardrails for my entire cluster before namespaces are even created"
+
+* > "As an OpenShift network administrator I want to enforce network traffic controls
+that are non-overridable by users in the cluster in order to secure my cluster"
+
+* > "As an OpenShift network administrator I want to enforce optional baseline network
+traffic controls that are overridable by users in the cluster if need be"
+
+Check out https://network-policy-api.sigs.k8s.io/user-stories/ where more fine grained
+user stories for ANP and BANP are outlined.
+
+### Goals
+
+* Provide support for admins to use Admin Network Policy API in OVN-Kubernetes
+* Provide support for admins to use Baseline Admin Network Policy API in OVN-Kubernetes
+* Provide support for admins to define `Pass` action rules in ANP API that will enable
```

Comment:
Confused that this is not part of the first goal? Is it downstream specific?

Reply:
oh actually you are right, this is part of first goal and its not downstream specific - just wanted to call out the extra shiny feature here but maybe that makes it more confusing. I will remove the 3rd goal.